packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -285,16 +304,38 @@ const useTransactionHistoryWithoutStatuses = (
     () => fetcher('withdrawals')
   )
 
-  const deposits = (depositsData || []).flat()
+  const deposits = [
+    ...getDepositsWithoutStatusesFromCache(),
+    (depositsData || []).flat()
+  ]
+
   const withdrawals = (withdrawalsData || []).flat()
 
   // merge deposits and withdrawals and sort them by date
-  const transactions = [...deposits, ...withdrawals, ...combinedCctpTransfers]
-    .flat()
-    .sort(sortByTimestampDescending)
+  const transactions = [
+    ...deposits,
+    ...withdrawals,
+    ...combinedCctpTransfers
+  ].flat()
+
+  // duplicates may occur when txs are taken from the local storage
+  const dedupedTransactions = useMemo(
+    () =>
+      Array.from(
+        new Map(
+          transactions.map(tx => [
+            `${tx.parentChainId}-${tx.childChainId}-${getTxIdFromTransaction(
+              tx
+            )}}`,
+            tx
+          ])
+        ).values()
+      ).sort(sortByTimestampDescending),
+    [transactions]
+  )
```

Comment:
use `Set` instead?

```suggestion
  const dedupedTransactions = useMemo(
    () => Array.from(new Set(transactions)).sort(sortByTimestampDescending),
    [transactions]
  )
```

Reply:
I think we may want to stick with Map. Set dedupes by reference and we have unrelated objects with the same data so we could end up with duplicates.

Reply:
as discussed on Slack, we should keep the Map implementation but leave a comment above to ensure it doesn't accidentally get refactored to Set